cmd/kube-scheduler/app/options/options.go

```diff
@@ -232,6 +232,10 @@ func emptySchedulerProfileConfig(profiles []kubeschedulerconfig.KubeSchedulerPro
 
 // Validate validates all the required options.
 func (o *Options) Validate() []error {
+	if o == nil {
```

Comment:
The only case `o` can be nil is:

https://github.com/kubernetes/kubernetes/blob/6d130d3b97242a6d9cfc2d9dc4d72030f441ec80/cmd/kube-scheduler/app/options/options.go#L75-L79

However, if `newDefaultComponentConfig` returns an error, it should have already been fatal at:

https://github.com/kubernetes/kubernetes/blob/ed3e0d302fb546653b78df583569b0311687a7a8/cmd/kube-scheduler/app/server.go#L64-L68

So this nil check looks unneeded.

Reply:
Thanks for review！

I've considered that some check are now unnecessary. But in fact, there are many such unnecessary checks in project. Such a check is always safer while we don't know when the code logic will change.